components/director/internal/domain/application/service.go

```diff
@@ -226,17 +229,22 @@ func (s *service) Create(ctx context.Context, in model.ApplicationRegisterInput)
 	}
 
 	id := s.uidService.Generate()
+	log.Debugf("ID %s generated for Application with name %s", id, in.Name)
+
 	app := in.ToApplication(s.timestampGen(), id, appTenant)
 
 	err = s.appRepo.Create(ctx, app)
 	if err != nil {
-		return "", err
+		return "", errors.Wrapf(err, "Error occurred while creating Application with name %s", in.Name)
 	}
 
+	log.Debugf("Ensuring Scenarios label definition exists for Tenant %s", appTenant)
 	err = s.scenariosService.EnsureScenariosLabelDefinitionExists(ctx, appTenant)
 	if err != nil {
 		return "", err
 	}
+
+	log.Debugf("Checking to add Default scenario if enabled for Application with id %s", id)
```

Comment:
well this does not bring any value. I'd be interested to see if the default scenario is actually added for this app or not. Telling that we will check does not bring much value